scm-diff-editor/src/lib.rs

```diff
@@ -439,12 +441,41 @@ fn print_dry_run(write_root: &Path, state: RecordState) {
     for file in files {
         let file_path = write_root.join(file.path.clone());
         let (selected_contents, _unselected_contents) = file.get_selected_contents();
-        match selected_contents {
-            SelectedContents::Absent => {
-                println!("Would delete file: {}", file_path.display())
-            }
+
+        let File {
+            file_mode: old_file_mode,
+            ..
+        } = file;
+
+        let SelectedChanges {
+            contents,
+            file_mode,
+        } = selected_contents;
+
+        if file_mode == FileMode::Absent {
+            println!("Would delete file: {}", file_path.display());
+            continue;
+        }
+
+        let mut printed_file_mode = false;
+        if old_file_mode != file_mode {
```

Comment:
nit: you can avoid the mutable variable and maybe make it more readable too like this:
```rust
let print_file_mode = old_file_mode != file_mode;
if print_file_mode {
...
}
```

Reply:
Good call, I've made that change